### PR TITLE
media-downloader: init at 2.4.0

### PR DESCRIPTION
**Pull request by zendo**
###### Description of changes

Media Downloader is a Qt/C++ front end to youtube-dl.

https://github.com/mhogomchungu/media-downloader

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/video/media-downloader/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  qt5,
```
Comment (viraptor) [inline pkgs/applications/video/media-downloader/default.nix:6]: The usual format is
```suggestion
{ lib
, stdenv
, fetchFromGitHub
, cmake
, qt5
```

File: pkgs/applications/video/media-downloader/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  nativeBuildInputs = [
+    cmake
+    qt5.wrapQtAppsHook
+  ];
+
+  meta = with lib; {
+    description = "A Qt/C++ GUI front end to youtube-dl";
+    homepage = "https://github.com/mhogomchungu/media-downloader";
+    license = with licenses; [ gpl2 gpl3 ];
```
Comment (viraptor) [inline pkgs/applications/video/media-downloader/default.nix:27]: https://github.com/mhogomchungu/media-downloader/blob/main/LICENSE.txt
I think it's [gpl2Plus](https://github.com/NixOS/nixpkgs/blob/aede85ad882a1b607d1820f211580cb3a4329080/lib/licenses.nix#L432)

File: pkgs/applications/video/media-downloader/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  };
+
+  nativeBuildInputs = [ cmake qt5.wrapQtAppsHook ];
+
+  preFixup = ''
+    qtWrapperArgs+=(
+    --prefix PATH : "${lib.makeBinPath [ ffmpeg-full aria2 yt-dlp python3 ]}"
+    )
+  '';
+
```
Comment (viraptor) [inline pkgs/applications/video/media-downloader/default.nix:30]: You could also add this to get automatic updates when a new version is tagged.

(needs `nix-update-script` in arguments too)
```suggestion

  passthru.updateScript = nix-update-script {
    attrPath = pname;
  };
```

**Review by viraptor — COMMENTED** [review]
This doesn't work on darwin. No window shows up and no messages are shown. Could be worth explicitly marking it as broken on `isDarwin` until someone can debug it.
Seems to be supported in general https://github.com/mhogomchungu/media-downloader/blob/386122552f58817d6e08de60db9b579569bd9bcc/src/utility.cpp#L109 and compiles successfully

File: pkgs/applications/video/media-downloader/default.nix (hunk 3 above)
Comment (zendo) [reply pkgs/applications/video/media-downloader/default.nix:30]: Darwin marked, maybe consider add `nix-update-script` next time.

File: pkgs/applications/video/media-downloader/default.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    repo = pname;
+    rev = "${version}";
+    sha256 = "sha256-EyfhomwBtdAt6HGRwnpiijm2D1LfaCAoG5qk3orDG98=";
+  };
+
+  nativeBuildInputs = [ cmake qt5.wrapQtAppsHook ];
+
+  preFixup = ''
+    qtWrapperArgs+=(
+    --prefix PATH : "${lib.makeBinPath [ ffmpeg-full aria2 yt-dlp python3 ]}"
```
Comment (AndersonTorres) [inline pkgs/applications/video/media-downloader/default.nix:27]:
```suggestion
      --prefix PATH : "${lib.makeBinPath [ ffmpeg-full aria2 yt-dlp python3 ]}"
```

Comment (AndersonTorres) [reply pkgs/applications/video/media-downloader/default.nix:27]: Silly but cute.

File: pkgs/applications/video/media-downloader/default.nix [hunk 5]
```diff
@@ -0,0 +1,39 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, qt5
+, ffmpeg-full
+, aria2
+, yt-dlp
```
Comment (AndersonTorres) [inline pkgs/applications/video/media-downloader/default.nix:8]: You said in the description it is a frontend for youtube-dl.
And here you are using an specific fork.
Any reason for that?

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/applications/video/media-downloader/default.nix (hunk 5 above)
Comment (zendo) [reply pkgs/applications/video/media-downloader/default.nix:8]: All description was copy from upstream, you can check the link see the detail.

Comment (AndersonTorres) [reply pkgs/applications/video/media-downloader/default.nix:8]: Understood.

**Review by AndersonTorres — APPROVED** [review]

**Pull request merged**